src/EnergyPlus/EconomicTariff.cc

```diff
@@ -3663,17 +3663,17 @@ void setNativeVariables(EnergyPlusData &state)
         // nativeTotalEnergy
         monthVal = 0.0;
         for (int kMonth = 1; kMonth <= NumMonths; ++kMonth) {
-            for (int jPeriod = 1; jPeriod <= (int)Period::Num; ++jPeriod) {
-                monthVal(kMonth) += tariff.gatherEnergy(kMonth)[jPeriod - 1];
+            for (int jPeriod = 0; jPeriod < (int)Period::Num; ++jPeriod) {
+                monthVal(kMonth) += tariff.gatherEnergy(kMonth)[jPeriod];
```

Comment:
@amirroth 's recommendation.

Mine would be to stop using an index based loop, so we'd never run into this problem to begin with.

STL accumulate:

```c++
for (int kMonth = 1; kMonth <= NumMonths; ++kMonth) {
    const auto& energy = tariff.gatherEnergy(kMonth);
    monthVal(kMonth) = std::accumulate(energy.begin(), energy.end(), 0.0);
}
```

Range-based for loop.

```c++
for (int kMonth = 1; kMonth <= NumMonths; ++kMonth) {
    for (const auto& v: tariff.gatherEnergy(kMonth)) {
        monthVal(kMonth) += v;
    }
}
```